compiler/src/dotty/tools/dotc/core/ConstraintHandling.scala

```diff
@@ -345,7 +345,7 @@ trait ConstraintHandling[AbstractContext] {
     else {
       val saved = constraint
       try
-        c2.forallParams(p =>
+        pre.forallParams(p =>
```

Comment:
This is now in contradiction to the doc comment. Also, I don't see how it could capture the intended meaning. 
If both `c1` and `c2` constrain different new type parameters that are not in `pre`, then according to the old meaning neither subsumes the other. But according to the new meaning each subsumes the other. This is clearly wrong, at least if we use the usual meaning of subsumes. 

This seems to be a critical problem that requires further thought to be addressed.